### DISCUSSION TITLE
Installation error

### DISCUSSION THREAD

**Discussion by mnishikawara** [PMEAL/porespy#995] [Q&A]
<img width="1920" alt="sc2024-10-28 105319" src="https://github.com/user-attachments/assets/1acdc99d-04e8-4cda-b287-b6790ba07a71">

I followed the installation guide by typing the following at the conda prompt: (conda install -c conda-forge porespy)
https://porespy.org/installation.html
After typing the script, the following error happened. The screenshot is attached.

--
Solving environment: \ warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package porespy-2.0.2-py310h5588dad_1 requires python >=3.10,<3.11.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ pin-1 is installable and it requires
│  └─ python 3.12.* , which can be installed;
└─ porespy is not installable because there are no viable options
   ├─ porespy [1.2.0|1.3.0|1.3.1] would require
   │  └─ python >=3.6,<3.7.0a0 , which conflicts with any installable versions previously reported;
   ├─ porespy [1.2.0|1.3.0|1.3.1|2.0.1|2.0.2] would require
   │  └─ python >=3.7,<3.8.0a0 , which conflicts with any installable versions previously reported;
   ├─ porespy [1.2.0|1.3.0|...|2.4.1] would require
   │  └─ python >=3.8,<3.9.0a0 , which conflicts with any installable versions previously reported;
   ├─ porespy [2.0.1|2.0.2|...|2.4.1] would require
   │  └─ python >=3.9,<3.10.0a0 , which conflicts with any installable versions previously reported;
   ├─ porespy [2.0.2|2.1|...|2.4.1] would require
   │  └─ python >=3.10,<3.11.0a0 , which conflicts with any installable versions previously reported;
   └─ porespy [2.3.0|2.4.1] would require
      └─ python >=3.11,<3.12.0a0 , which conflicts with any installable versions previously reported.
--

How can I solne the problem?

Masa

**Comment by jgostick** [comment]
Not totally sure what is causing your problem, but seems like you have too many packages in your environment and they are starting to conflict with each other.  I suggest you make a new fresh env and install python. 

HOWEVER, be sure to use python <3.13 since `llvmlite` does not seem to work in 3.13 yet, and `numba` requires it.  I"m sure they'll get this sorted out soon, but as of yesterday I needed to use python 3.12.5.

**Reply by mnishikawara** [reply]
Thank you. 
By the way, What is the relationship between porespy and openpnm?
I had the installation problem for openpnm, but it was solved.
https://github.com/PMEAL/OpenPNM/discussions/2941
If openpnm can cover porespy, I don't need to install porespy.

**Comment by jgostick** [comment]
You need to open Spyder or VSCode, or perhaps use JupyterLab.  Then you can write a file like:

```python
import porespy as ps
import matplotlib.pyplot as plt
im = ps.generators.blobs(shape=[500, 500], porosity=0.6, blobiness=2)
plt.imshow(im)
```

**Comment by jgostick** [comment]
Basically, the 'conda' prompt is for installing things, while the IDE (Spyder, Jupyter, etc) is where you run the code.

**Reply by mnishikawara** [reply]
Thank you so much. I could run the example.
